### PR TITLE
ROX-13938: Migrate shell unit tests

### PR DESCRIPTION
**Pull request by janisz**
## Description

Migrate shell unit test and adjust code to work on GHA.

### CODE REVIEW

**Comment by janisz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #4127** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/4127" target="_blank">Graphite</a> 
    * **PR #4143** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/4143" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/4143?utm_source=stack-comment).

File: scripts/ci/bats/lib_get_repo_full_name.bats [hunk 1]
```diff
@@ -17,6 +19,21 @@ function setup() {
...
 }
 
+@test "GITHUB_ACTION but nothing else" {
+    export GITHUB_ACTION=true
+    run get_repo_full_name
+    assert_failure 1
+    assert_output --partial 'expect: GITHUB_ACTION_REPOSITORY'
+}
+
+@test "GITHUB_ACTION and " {
```
Comment (msugakov) [inline scripts/ci/bats/lib_get_repo_full_name.bats:29]: This test name looks unfinished. Perhaps
```suggestion
@test "GITHUB_ACTION and GITHUB_ACTION_REPOSITORY" {
```
?

File: .github/workflows/unit-tests.yaml [hunk 2]
```diff
@@ -86,3 +86,16 @@ jobs:
...
+  shell:
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/stackrox-io/apollo-ci:stackrox-test-0.3.50
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        ref: ${{ github.event.pull_request.head.sha }}
```
Comment (msugakov) [inline .github/workflows/unit-tests.yaml:99]: Similar to https://github.com/stackrox/stackrox/pull/4127#discussion_r1053243794 , maybe just omit the `ref`?

**Review by msugakov — COMMENTED** [review]
Looks ok with two nitpicks.

**Pull request closed without merging**